TS 4.8.2: type error

From bumping typescript to [4.8.2](https://devblogs.microsoft.com/typescript/announcing-typescript-4-8/) in my app that is using this library:

```
node_modules/snakecase-keys/index.d.ts:54:39 - error TS2344: Type 'T[P]' does not satisfy the constraint 'readonly any[] | Record<string, any>'.
  Type 'T[keyof T]' is not assignable to type 'readonly any[] | Record<string, any>'.
    Type 'T[string] | T[number] | T[symbol]' is not assignable to type 'readonly any[] | Record<string, any>'.
      Type 'T[string]' is not assignable to type 'readonly any[] | Record<string, any>'.
        Type 'T[string]' is not assignable to type 'readonly any[]'.
          Type 'T[keyof T]' is not assignable to type 'readonly any[]'.
            Type 'T[P]' is not assignable to type 'readonly any[]'.
              Type 'T[keyof T]' is not assignable to type 'readonly any[]'.
                Type 'T[string] | T[number] | T[symbol]' is not assignable to type 'readonly any[]'.
                  Type 'T[string]' is not assignable to type 'readonly any[]'.

54         [P in keyof T]: SnakeCaseKeys<T[P], Deep, Exclude>;
```

--- a/index.d.ts
+++ b/index.d.ts
@@ -51,7 +51,9 @@ declare namespace snakecaseKeys {
   > = T extends readonly any[]
     ? // Handle arrays or tuples.
       {
-        [P in keyof T]: SnakeCaseKeys<T[P], Deep, Exclude>;
+        [P in keyof T]: T[P] extends Record<string, any> | readonly any[]
+        ? SnakeCaseKeys<T[P], Deep, Exclude>
+        : T[P];
       }
     : T extends Record<string, any>
     ? // Handle objects.

--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
   "devDependencies": {
     "standard": "^17.0.0",
     "tape": "^5.0.1",
-    "tsd": "^0.22.0"
+    "tsd": "^0.23.0"
   },
   "files": [
     "index.js",

